apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx

```diff
@@ -0,0 +1,266 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { AlertInterface, AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { Field, Form } from "@wso2is/form";
+import { ConfirmationModal, DangerZone, DangerZoneGroup, EmphasizedSegment } from "@wso2is/react-components";
+import { AxiosResponse } from "axios";
+import React, { FunctionComponent, ReactElement, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import { Divider } from "semantic-ui-react";
+import { AppConstants, history } from "../../../core";
+import { deleteRoleById, searchRoleList, updateRoleDetails } from "../../api";
+import { RoleConstants } from "../../constants";
+import { PatchRoleDataInterface, RoleEditSectionsInterface, SearchRoleInterface } from "../../models/roles";
+
+/**
+ * Interface to contain props needed for component
+ */
+type BasicRoleProps = IdentifiableComponentInterface & RoleEditSectionsInterface;
+
+const FORM_ID: string = "edit-role-basic";
+
+/**
+ * Component to edit basic role details.
+ *
+ * @param props - Role object containing details which needs to be edited.
+ */
+export const BasicRoleDetails: FunctionComponent<BasicRoleProps> = (props: BasicRoleProps): ReactElement => {
```

Comment:
We already have a component with the same name. That will be replaced with this in the future?

Reply:
Yes, that will be replaced with this in a future PR.